// @vuepress/core/lib/pluginApi/createPluginApiHooks.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createPluginApiHooks = void 0;
const createHookQueue_1 = require("./createHookQueue");
const createPluginApiHooks = () => ({
    // life cycle hooks
    onInitialized: (0, createHookQueue_1.createHookQueue)('onInitialized'),
    onPrepared: (0, createHookQueue_1.createHookQueue)('onPrepared'),
    onWatched: (0, createHookQueue_1.createHookQueue)('onWatched'),
    onGenerated: (0, createHookQueue_1.createHookQueue)('onGenerated'),
    // page hooks
    extendsPageOptions: (0, createHookQueue_1.createHookQueue)('extendsPageOptions'),
    extendsPageData: (0, createHookQueue_1.createHookQueue)('extendsPageData'),
    // markdown hooks
    extendsMarkdown: (0, createHookQueue_1.createHookQueue)('extendsMarkdown'),
    // client files hooks
    clientAppEnhanceFiles: (0, createHookQueue_1.createHookQueue)('clientAppEnhanceFiles'),
    clientAppRootComponentFiles: (0, createHookQueue_1.createHookQueue)('clientAppRootComponentFiles'),
    clientAppSetupFiles: (0, createHookQueue_1.createHookQueue)('clientAppSetupFiles'),
    // bundler hooks
    alias: (0, createHookQueue_1.createHookQueue)('alias'),
    define: (0, createHookQueue_1.createHookQueue)('define'),
});
exports.createPluginApiHooks = createPluginApiHooks;
